cpp/merge_module/algorithm/merge.cpp

```diff
@@ -1 +1,90 @@
-#include "merge.hpp"
\ No newline at end of file
+#include "merge.hpp"
+
+bool Merge::LabelsContained(const std::unordered_set<std::string_view> &labels, const mgp::Node &node) {
+  bool contained = false;
+  auto size = labels.size();  // this works if labels are unique, which they are
+  size_t counter = 0;
+
+  for (const auto label : node.Labels()) {
+    if (labels.find(label) != labels.end()) {
+      counter++;
+    }
+  }
+
+  if (counter == size) {
+    contained = true;
+  }
+
+  return contained;
+}
+
+bool Merge::IdentProp(const mgp::Map &identProp, const mgp::Node &node) {
+  bool match = true;
+  for (const auto &[key, value] : identProp) {
+    if (value != node.GetProperty(std::string(key))) {
+      match = false;
+    }
+  }
+  return match;
+}
```

Comment:
I suggest using all_of instead of for loop, furthermore I suggest this snippet of my code for merge relationships:
```template <typename T>
concept GraphObject = std::is_same<T, mgp::Node>::value || std::is_same<T, mgp::Relationship>::value;

template <GraphObject NodeOrRel>
bool SameProps(const NodeOrRel &node_or_rel, const mgp::Map &props) {
  return std::all_of(props.begin(), props.end(), [&node_or_rel](auto &map_item) {
    return node_or_rel.GetProperty(std::string(map_item.key)) == map_item.value;
  });
}
```
Since this is also used in merge.relationship we can use the same function for checking properties, with concepts we can ensure that GraphObject is of type mgp::Node or mgp::Relationship

Reply:
My bad, this is an old problem and will cause a memory leak, new suggestion for SameProps
```
for (const auto &[k, v] : props) {
    if (node_or_rel.GetProperty(std::string(k)) != v)
      return false;
  }
  return true;
```